opensearchpy/connection/http_requests.py

```diff
@@ -187,9 +189,17 @@ def perform_request(  # type: ignore
             "allow_redirects": allow_redirects,
         }
         send_kwargs.update(settings)
+
+        calculate_service_time = False
```

Comment:
This seems very specific. I propose to add `metrics` parameter that would default to `None` or an instance of `Metrics`. I'd like to be able to write 

```py
client = Client(metrics = Metrics())
```

Then the `if` is not necessary.